domain/consensus/processes/syncmanager/syncmanager.go

```diff
@@ -69,6 +70,32 @@ func (sm *syncManager) GetHashesBetween(stagingArea *model.StagingArea, lowHash,
 	return sm.antiPastHashesBetween(stagingArea, lowHash, highHash, maxBlocks)
 }
 
+func (sm *syncManager) GetPastDiff(stagingArea *model.StagingArea, hasHash,
+	requestedHash *externalapi.DomainHash, maxBlocks uint64) (hashes []*externalapi.DomainHash, err error) {
+	onEnd := logger.LogAndMeasureExecutionTime(log, "GetPastDiff")
+	defer onEnd()
+	isHasAncestorOfRequested, err := sm.dagTopologyManager.IsAncestorOf(stagingArea, hasHash, requestedHash)
+	if err != nil {
+		return nil, err
+	}
+	if isHasAncestorOfRequested {
+		return nil, errors.Errorf("expected block %s to be in anticone of %s",
+			hasHash,
+			requestedHash)
+	}
+	isRequestedAncestorOfHas, err := sm.dagTopologyManager.IsAncestorOf(stagingArea, requestedHash, hasHash)
+	if err != nil {
+		return nil, err
+	}
+	if isRequestedAncestorOfHas {
+		return nil, errors.Errorf("expected block %s to be in anticone of %s",
+			hasHash,
+			requestedHash)
+	}
+	// TODO: use maxBlocks limit
```

Comment:
This must be handled before merge

Reply:
How do you suggest doing it? Should I propegate a max lim into AnticoneFromBlock?

Reply:
I mean, I don't want to enter a long traversal if the syncee is dishonest